dipy/viz/ui.py

```diff
@@ -3303,3 +3306,225 @@ def left_button_clicked(self, i_ren, obj, list_box_item):
+                              self.scroll_callback)
+            self.add_callback(slot.textblock.actor, up_event,
+                              self.scroll_callback)
+            self.add_callback(slot.textblock.actor, down_event,
+                              self.scroll_callback)
+            slot.add_callback(slot.textblock.actor, "LeftButtonPressEvent",
+                              self.directory_click_callback)
+            slot.add_callback(slot.background.actor, "LeftButtonPressEvent",
+                              self.directory_click_callback)
+
+    def _get_actors(self):
+        """ Get the actors composing this UI component.
+        """
+        return self.listbox.actors
+
+    def resize(self, size):
+        pass
+
+    def _set_position(self, coords):
+        """ Position the lower-left corner of this UI component.
+        Parameters
+        ----------
+        coords: (float, float)
+            Absolute pixel coordinates (x, y).
+        """
+        self.listbox.position = coords
+
+    def _add_to_renderer(self, ren):
+        """ Add all subcomponents or VTK props that compose this UI component.
+        Parameters
+        ----------
+        ren : renderer
+        """
+        self.listbox.add_to_renderer(ren)
+
+    def _get_size(self):
+        return self.listbox.size
+
+    def get_all_file_names(self):
+        """ Gets file and directory names.
+        Returns
+        -------
+        all_file_names: list((string, {"directory", "file"}))
+            List of all file and directory names as string.
+        """
+        all_file_names = []
+
+        directory_names = self.get_directory_names()
+        for directory_name in directory_names:
+            all_file_names.append((directory_name, "directory"))
+
+        file_names = self.get_file_names()
+        for file_name in file_names:
+            all_file_names.append((file_name, "file"))
+
+        return all_file_names
+
+    def get_directory_names(self):
+        """ Finds names of all directories in the current_directory
+        Returns
+        -------
+        directory_names: list(string)
+            List of all directory names as string.
+        """
+        # A list of directory names in the current directory
+        directory_names = []
+        for (_, dirnames, _) in os.walk(self.current_directory):
+            directory_names += dirnames
+            break
+        directory_names.sort(key=lambda s: s.lower())
+        directory_names.insert(0, "../")
+        return directory_names
+
+    def get_file_names(self):
+        """ Finds names of all files in the current_directory
+        Returns
+        -------
+        file_names: list(string)
+            List of all file names as string.
+        """
+        # A list of file names with extension in the current directory
+        for (_, _, files) in os.walk(self.current_directory):
+            break
+
+        file_names = []
+        if "*" in self.extensions:
```

Comment:
It should also probably treat an empty string "" the same as "*". Then it can accept `None` for directories-only.